### PR TITLE
PLAT-10535: Extension App RSA authentication

### PR DESCRIPTION
**Pull request by symphony-elias**
### Ticket
PLAT-10535

### Description
Implemented extension app authentication

### Checklist
- [x] Referenced a ticket in the PR title and in the corresponding section
- [x] Filled properly the description and dependencies, if any
- [x] Unit tests updated or added
- [x] Docstrings added or updated
- [x] Updated the documentation in [docs folder](../docs)

### CODE REVIEW

File: symphony/bdk/core/auth/auth_session.py [hunk 1]
```diff
@@ -96,4 +102,57 @@ async def session_token(self):
...
+        """Triggers re-authentication to refresh the tokens.
+
+        :return: None
+        """
+        app_tokens = await self._authenticator.authenticate_and_retrieve_tokens(self._app_token)
+        self._symphony_token = app_tokens.symphony_token
+        self._app_token = app_tokens.app_token
+        self._expire_at = app_tokens.expire_at
+
+    def symphony_token(self) -> str:
```
Comment (thibauult) [inline symphony/bdk/core/auth/auth_session.py:140]: Question mainly for myself :) No need to annotate with `@property` in this case?

Comment (symphony-elias) [reply symphony/bdk/core/auth/auth_session.py:140]: We could. But IMHO `@property` is really useful when we need getters and setters with specific logic inside. With no setter, added value is less obvious.

Comment (symphony-elias) [reply symphony/bdk/core/auth/auth_session.py:140]: Done in https://github.com/SymphonyPlatformSolutions/symphony-api-client-python/pull/153/commits/423447eeb207ba23964499cd4d9df8f16708bba8

File: docsrc/markdown/authentication.md [hunk 2]
```diff
@@ -91,6 +91,27 @@ app:
       path: /path/to/rsa/private-key.pem
 ```
 
+### Circle Of Trust
+> Read more about Circle Of Trust
```
Comment (symphony-youri) [inline docsrc/markdown/authentication.md:95]: maybe just explain that for CoT you would have to implement endpoints rely on app_authenticator ?
and that you are free to choose a web framework

File: symphony/bdk/core/auth/jwt_helper.py [hunk 3]
```diff
@@ -1,25 +1,73 @@
...
 
+from jose import jwt, JOSEError
 
-def create_signed_jwt(private_key_config, username, expiration=None):
+from symphony.bdk.core.auth.exception import AuthInitializationError
+from symphony.bdk.core.config.model.bdk_rsa_key_config import BdkRsaKeyConfig
+
+RS512 = "RS512"
+
+DEFAULT_EXPIRATION = 290
```
Comment (symphony-youri) [inline symphony/bdk/core/auth/jwt_helper.py]: Maybe put in here again the explanation on why this value?
or write 5 * 60 - 10? that would help understand it is 4min 50s

File: symphony/bdk/core/auth/jwt_helper.py [hunk 4]
```diff
@@ -1,25 +1,73 @@
...
+from cryptography.hazmat.primitives.serialization import Encoding, PublicFormat
+from cryptography.x509 import load_pem_x509_certificate
 
+from jose import jwt, JOSEError
 
-def create_signed_jwt(private_key_config, username, expiration=None):
+from symphony.bdk.core.auth.exception import AuthInitializationError
+from symphony.bdk.core.config.model.bdk_rsa_key_config import BdkRsaKeyConfig
+
+RS512 = "RS512"
```
Comment (symphony-youri) [inline symphony/bdk/core/auth/jwt_helper.py]: maybe name it JWT_ENCRYPTION? (in the end RS512 is just a detail)

**Review by symphony-youri — APPROVED** [review]

**Pull request merged**